docusaurus/docs/what-is-included/internationalization.md

```diff
@@ -4,4 +4,32 @@ title: Internationalization
 sidebar_label: Internationalization
 ---
 
-## TBD
\ No newline at end of file
+All of our projects have i18n support built-in, even if there's no need to initially have more than one locale. By having i18n support from the very beginning, the project itself is built with that in mind, making it very easy to add new locales in the future without having to refactor a good surface of your app.
+
+We have chosen [`react-intl`](https://github.com/formatjs/react-intl/) for internationalization and localization because:
+
+- It has a large community and is being actively maintained.
+- Built on the standard [Intl](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl) API.
+- It can localize strings, numbers, dates and relative dates.
+- Runs in the browser, Node.js and React Native.
+
+Additionally, we also make use of [`react-native-localize`](https://github.com/react-native-community/react-native-localize) to get hold of locale settings from the device's operating system.
+
+## Locale resolution
+
+When the app is launched, if `initialLocaleId` is not defined, `LocaleProvider` attempts to find the best possible match between the preferred locales defined in the device's operating system and the locales defined for the app. If a match is found, the translations for the matched locale are loaded. Otherwise, the translations for the default locale, `en-US`, are loaded. If your app offers an in-app locale picker and persists the user's choice, i.e. with Redux, make sure to pass it to `initialLocaleId` when `LocaleProvider` is first rendered.
+
+## Adding a new locale
+
+1. Add the locale to the `intl/index.js` file, following the default `en-US` locale.
+2. Add the messages file to `intl/messages/<locale>.json`.
+
+## Removing this feature
+
+If you are really sure internationalization is not needed in your project, you'll want to remove all the unnecessary `intl` related code. Be sure to follow these steps in order to clean your project properly:
+
+1. Uninstall `react-intl`, `react-native-localize` and `full-icu`.
+2. Remove the `NODE_ICU_DATA` environment variable from `test` script, found in `package.json`.
+3. Remove the `intl` folder.
+4. Search globally for `react-intl` and remove the corresponding code across the project.
```

Comment:
Is it so widespread throughout the project that at least the files can't be listed here? 

IMHO, it seems a bit off when a guide asks me to find all and remove because that's what I would do if there was no guide 🤔

Reply:
We're talking about stuff like this:

```js
import { FormattedMessage } from 'react-intl';
```

which is used in several components. I don't really mind having this instruction explicit in the guide as no harm is done anyway, even if it feels some what redundant because it is implicit. Worst case scenario, you will hit a "module not found" error but this last step anticipates just that. 😉

Reply:
I think you're misinterpreting, my intention is exactly to make this as explicit as possible, that's why I suggested listing the files where you would need to delete that stuff!

Either way, it's fine 😄

Reply:
I get it. The only issue I see is if the boilerplate changes, i.e. we add or remove a component which uses `FormattedMessage` or other APIs, we'd have to keep the documentation in sync.

Reply:
True, that can be a bit of a hassle. Maybe leave it as is?

Reply:
Agreed!